app/src/main/java/com/cornellappdev/android/eatery/ExpandableListAdapter.java

```diff
@@ -101,67 +107,43 @@ public View getGroupView(int i, boolean b, View view, ViewGroup viewGroup) {
       view = infalInflater.inflate(R.layout.list_view_header, viewGroup, false);
     }
 
-    String m = (String) getGroup(i);
+    DiningHallModel dm = (DiningHallModel) getGroup(i);
+    String diningName = (dm).getNickName();
     TextView headerText = view.findViewById(R.id.header);
-    headerText.setText(m);
-    headerText.setTypeface(null, Typeface.NORMAL);
-
-    TextView timetext1 = view.findViewById(R.id.time1);
-
-    try {
-      CafeteriaModel myCafe = null;
-      int count = 0;
-      while (count < cafeData.size()) {
-        if (m.equals(cafeData.get(count).getNickName())) {
-          myCafe = cafeData.get(count);
+    headerText.setText(diningName);
+    headerText.setTypeface(null, Typeface.BOLD);
+
+    TextView openText = view.findViewById(R.id.weekly_open);
+    TextView timeText = view.findViewById(R.id.weekly_time);
+
+    LocalDateTime currentTime = LocalDateTime.now();
+    boolean set = false;
+    Interval interval = dm.getIntervalByDateAndType(date,mealType);
+    if(interval == null)
+    {
+      interval = dm.getIntervalByDateAndType(date,BRUNCH);
+      mealType = BRUNCH;
+    }
+    if(interval!=null) {
```

Comment:
same thing, there should be spaces surrounding `!=` as well. Lastly, if there's a lot of logic for `if (interval != null)`, then short-circuit:

```
if (interval == null) return;
/* if interval != null code goes here, less indentation */
```

Reply:
fixed